### ISSUE TITLE
Crash on invalid weight calculation

### ISSUE THREAD

**Issue by HoveringHalibut** [datawire/ambassador#1289]
**Describe the bug**
When 3 services have the same route annotations with no weight specified, the calculated result is not a valid integer and throws the following error:
Unable to parse JSON as proto (INVALID_ARGUMENT:(route_config.virtual_hosts[0].routes[23].route.weighted_clusters.clusters[2].weight.value): invalid value 33.333333333333336 for type TYPE_UINT32)

I assume this would happen on any weight calculation with a remainder.

**To Reproduce**
Deploy 3 services with the same route annotation (this a sanitized version the service that caused the error):
Annotations:       
    getambassador.io/config:
        ---
        apiVersion: ambassador/v0
        kind:  Mapping
        name:  "myservice_mapping"
        prefix: "/"
        service: "myservice-ambassador"
        rewrite: "/"
        host: "test.example.com"

**Expected behavior**
The weight is rounded to a valid integer.

**Versions (please complete the following information):**
 - Ambassador: 0.50.3
 - Kubernetes environment: AKS
 - Version: 1.12.5

**Additional context**
I'm not intending to use a configuration like this, it was the result of an issue with our deployment pipeline.

**Comment by jonathanbeber** [comment]
Just got the same error in `0.52.0`, apparently it still happening. I missed something on my previous tests on #813 

```
error initializing configuration \'/ambassador/snapshots/econf-tmp.json\': Unable to parse JSON as proto (INVALID_ARGUMENT:(route_config.virtual_hosts[0].routes[4].route.weighted_clusters.clusters[2].weight.value): invalid value 33.333333333333336 for type TYPE_UINT32
```

**Comment by esmet** [comment]
I'm currently testing a fix and will submit a PR soon.

**Comment by kflynn** [comment]
Fixed in #1494. Thanks!

**Comment by kflynn** [comment]
Shipped in 0.61.
